Mathlib/Data/Fin/Tuple/Sort.lean

```diff
@@ -107,6 +108,20 @@ theorem monotone_sort (f : Fin n → α) : Monotone (f ∘ sort f) := by
   exact (monotone_proj f).comp (graphEquiv₂ f).monotone
 #align tuple.monotone_sort Tuple.monotone_sort
 
+/-- All the elements `· ≤ a` appear the start of a sorted tuple -/
+theorem sort_lt_at_start_of_monotone {α} [LinearOrder α] (m : ℕ) (f : Fin m → α) (a : α)
+    (h_sorted : Monotone f)
+    (j : Fin m) (h : j < Fintype.card {i // f i ≤ a}) :
+    f j ≤ a := by
+  contrapose! h
+  have := Fintype.card_subtype_compl (¬f · ≤ a)
+  simp_rw [not_not, not_le] at this
+  rw [this, Fintype.card_fin, tsub_le_iff_tsub_le, Fintype.card_subtype]
+  refine le_trans (by simp) (Finset.card_mono $ show Finset.Ici j ≤ _ from fun k hk ↦ ?_)
+  simp only [Finset.mem_Ici] at hk
+  exact Finset.mem_filter.mpr ⟨Finset.mem_univ _, h.trans_le (h_sorted hk)⟩
```

Comment:
somewhat shorter, using `Iio` rather than `Ici`:
```suggestion
  rw [← Fin.card_Iio, Fintype.card_subtype]
  refine Finset.card_mono (fun i => Function.mtr ?_)
  simp_rw [Finset.mem_filter, Finset.mem_univ, true_and, Finset.mem_Iio]
  exact fun hij => (h.trans_le <| h_sorted <| le_of_not_lt hij).not_le
```

Reply:
Thanks I replaced the old one